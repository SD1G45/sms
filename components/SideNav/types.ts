export interface SideNavProps extends React.HTMLAttributes<HTMLDivElement> {
  items: string[];
  routes: string[];
  heading: string;
}
