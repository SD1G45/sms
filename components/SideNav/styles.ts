import styled from "styled-components";

export const StyledDiv = styled.div`
  margin-left: 10px;
  padding-top: 40px;
  margin-right: 10px;
`;

export const StyledTable = styled.table`
  border-collapse: collapse;
`;

export const Item = styled.td<{ active: boolean }>`
  border-left: 2.75px solid ${(props) => (props.active ? "#4881F0" : "#C4C4C4")};
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 10px;
  cursor: pointer;
  white-space: nowrap;
`;

export const Header = styled.h1`
  font-size: 1.3rem;
  font-weight: bold;
  margin-bottom: 17.5px;
`;
