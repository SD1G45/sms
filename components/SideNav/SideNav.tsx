import React from "react";
import { StyledTable, StyledSelected, StyledNonSelected, Header, StyledDiv } from "./styles";
import { SideNavProps } from "./types";

const SideNav: React.FC<SideNavProps> = ({
    items,
    index,
    heading
}) => {
    return (
        <StyledDiv>
            <Header>{heading}</Header>
                <StyledTable>
                    {items.map((item, i) => i === index ? 
                        <tr key={i}><StyledSelected>{item}</StyledSelected></tr>
                        :
                        <tr key={i}><StyledNonSelected>{item}</StyledNonSelected></tr>
                    )}
                </StyledTable>
        </StyledDiv>);
};

export default SideNav;