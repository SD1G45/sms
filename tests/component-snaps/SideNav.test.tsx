import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import SideNav from "../../components/SideNav";

const items: string[] = ["Analytics", "Create New", "FAQ"];
const routes: string[] = ["/test", "/test", "/test"];

jest.mock("next/router", () => ({
  useRouter() {
      return {
          route: "/",
          pathname: "",
          query: "",
          asPath: "",
      };
  },
}));

describe("<SideNav/>", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SideNav items={items} routes={routes} heading={"Sample"}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});