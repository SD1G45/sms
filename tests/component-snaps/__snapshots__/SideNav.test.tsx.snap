// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNav/> renders correctly 1`] = `
.c0 {
  margin-left: 10px;
  padding-top: 40px;
}

.c3 {
  border-collapse: collapse;
}

.c4 {
  border-left: 2.75px solid #C4C4C4;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 10px;
  cursor: pointer;
  white-space: nowrap;
}

.c2 {
  font-size: 1.3rem;
  font-weight: bold;
  margin-bottom: 17.5px;
}

.c1 {
  width: 100px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      Sample
    </h1>
  </div>
  <table
    className="c3"
  >
    <tbody>
      <tr
        href="/test"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <td
          className="c4"
          id="Analytics"
        >
          Analytics
        </td>
      </tr>
      <tr
        href="/test"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <td
          className="c4"
          id="Create New"
        >
          Create New
        </td>
      </tr>
      <tr
        href="/test"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <td
          className="c4"
          id="FAQ"
        >
          FAQ
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
